pkg/controllers/utils.go

```diff
@@ -0,0 +1,69 @@
+package controllers
```

Comment:
We can add these functions to the structs themselves as methods and forego the util file.

Reply:
This util file has to/fro conversion between k8s client structs and cloudmap's representation. I wanted to keep this layer separate.